pkg/controller/useraccount/useraccount_controller.go

```diff
@@ -301,21 +326,19 @@ func (r *ReconcileUserAccount) addFinalizer(userAcc *toolchainv1alpha1.UserAccou
 	return nil
 }
 
-// manageCleanUp deletes the identity, user and finalizer when the UserAccount is being deleted
-func (r *ReconcileUserAccount) manageCleanUp(userAcc *toolchainv1alpha1.UserAccount) error {
+// deleteUserAndIdentity deletes the identity and user. Returns bool and error indicating that
+// whether the user/identity were deleted.
+func (r *ReconcileUserAccount) deleteUserAndIdentity(userAcc *toolchainv1alpha1.UserAccount) (bool, error) {
+
 	if deleted, err := r.deleteIdentity(userAcc); err != nil || deleted {
```

Comment:
It's confusing. You return `deleted:false` if the identity was deleted and there is no error?